### PR TITLE
Archetype for Global Shared Library with pipeline unit tests

### PR DESCRIPTION
**Pull request by ozangunalp**
TODO
- [x] Archetype description -> TODO
- [x] Needs #22
- [x] Default value for package -> org.jenkinsci.pipeline.sample

### CODE REVIEW

File: global-shared-library/src/main/resources/META-INF/maven/archetype-metadata.xml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<archetype-descriptor xmlns="http://maven.apache.org/plugins/maven-archetype-plugin/archetype-descriptor/1.0.0"
+                      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                      xsi:schemaLocation="http://maven.apache.org/plugins/maven-archetype-plugin/archetype-descriptor/1.0.0 http://maven.apache.org/xsd/archetype-descriptor-1.0.0.xsd"
+                      name="Global Shared Library Example">
+    <requiredProperties>
+        <requiredProperty key="groupId">
+            <defaultValue>unused</defaultValue>
+        </requiredProperty>
+        <requiredProperty key="package">
+            <defaultValue>org.jenkinsci.pipeline.sample</defaultValue>
```
Comment (jglick) [inline global-shared-library/src/main/resources/META-INF/maven/archetype-metadata.xml]: In `archetype.properties` you claim this is ignored. Either it _is_ ignored, in which case the `defaultValue` here should say so; or it is actually used, in which case the test instantiation should use this default.

File: global-shared-library/src/main/resources/META-INF/maven/archetype-metadata.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    </requiredProperties>
+    <fileSets>
+        <fileSet filtered="true">
+            <directory>.</directory>
+            <includes>
+                <include>pom.xml</include>
+            </includes>
+        </fileSet>
+        <fileSet filtered="true">
+            <directory></directory>
```
Comment (jglick) [inline global-shared-library/src/main/resources/META-INF/maven/archetype-metadata.xml]: `.`?

File: global-shared-library/src/main/resources/META-INF/maven/archetype-metadata.xml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            <includes>
+                <include>README.md</include>
+            </includes>
+        </fileSet>
+        <!-- Shared Library sources -->
+        <fileSet packaged="true">
+            <directory>shared-library/resources</directory>
+            <includes>
```
Comment (jglick) [inline global-shared-library/src/main/resources/META-INF/maven/archetype-metadata.xml]: I think can be omitted if you are just including everything.

**Comment by ozangunalp** [comment]
Build seems to fail during `Empty Jenkins Plugin` with a `ReasonPhrase:Gateway Time-out.`

File: global-shared-library/src/main/resources/archetype-resources/shared-library/src/Utils.groovy [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package $package
+
+class Utils implements Serializable {
+
+	def script
+
+	Utils() {
```
Comment (jglick) [inline global-shared-library/src/main/resources/archetype-resources/shared-library/src/Utils.groovy]: Unused?

File: global-shared-library/src/main/resources/archetype-resources/shared-library/vars/acme.groovy [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// vars/acme.groovy
+class acme implements Serializable {
```
Comment (jglick) [inline global-shared-library/src/main/resources/archetype-resources/shared-library/vars/acme.groovy]: BTW I do not actually recommend ever declaring types in `vars/*.groovy`. Should just delete the part of the docs that discusses it.

File: global-shared-library/src/main/resources/archetype-resources/unit-tests/src/main/jenkins/pipelineUsingSharedLib.jenkins [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package $package
```
Comment (jglick) [inline global-shared-library/src/main/resources/archetype-resources/unit-tests/src/main/jenkins/pipelineUsingSharedLib.jenkins]: Any reason not to use `.groovy` extension?

**Comment by jglick** [comment]
Seems we have a test failure…

File: global-shared-library/src/main/resources/META-INF/maven/archetype-metadata.xml (hunk 1 above)
Comment (ozangunalp) [reply global-shared-library/src/main/resources/META-INF/maven/archetype-metadata.xml]: Indeed, I'll change the groupId to `org.sample` and make changes in the `archetype.properties`. Same for `scripted-pipeline` btw.

File: global-shared-library/src/main/resources/META-INF/maven/archetype-metadata.xml (hunk 2 above)
Comment (ozangunalp) [reply global-shared-library/src/main/resources/META-INF/maven/archetype-metadata.xml]: It is odd but : https://stackoverflow.com/a/15435608

File: global-shared-library/src/main/resources/archetype-resources/shared-library/vars/acme.groovy (hunk 5 above)
Comment (ozangunalp) [reply global-shared-library/src/main/resources/archetype-resources/shared-library/vars/acme.groovy]: Then it is better to delete it from here.

**Comment by ozangunalp** [comment]
Currently projects inside the `pipeline-unit` profile will fail after every release. See 7819ade193ff76a3700bf8926487bcc103d28456

File: pom.xml [hunk 7]
```diff
@@ -101,6 +101,7 @@
             <id>pipeline-unit</id>
             <modules>
                 <module>scripted-pipeline</module>
+                <module>global-shared-library</module>
```
Comment (jglick) [inline pom.xml]: Do not forget to get these moved into the main section and the `profile` deleted once hosting is complete!

**Pull request merged**